src/navbar/navbar-items.tsx

```diff
@@ -52,30 +121,72 @@ export const NavbarItems = <T,>({
-                        data-testid={testId}
-                        weight={textWeight}
-                        $selected={selected} /* for mobile */
-                        {...otherItemAttrs}
-                        onClick={handleLinkClick(item)}
-                        {...options}
-                    >
-                        {children}
-                        {selected && <LinkIndicator />}
-                    </Link>
+                    <MenuItemContainer>
+                        <Link
+                            data-testid={testId}
+                            weight={textWeight}
+                            $selected={selected} /* for mobile */
+                            {...otherItemAttrs}
+                            onClick={handleLinkClick(item, index)}
+                            {...options}
+                        >
+                            {children}
+                            {selected && <LinkIndicator />}
+                            {isMobile && item.subMenu && (
+                                <MenuItemRightContainer>
+                                    {selected ? <UpIcon /> : <DownIcon />}
+                                </MenuItemRightContainer>
+                            )}
+                        </Link>
+                    </MenuItemContainer>
+
+                    {isMobile &&
+                        selectedIndex >= 0 &&
+                        selectedIndex === index &&
+                        toggleDropdown && (
+                            <Menu
+                                items={item.subMenu}
+                                selectedId={selectedId}
+                                mobile={isMobile}
+                                onItemClick={handleSubLinkClick(
+                                    item?.subMenu,
+                                    0
+                                )}
+                            ></Menu>
+                        )}
                 </LinkItem>
             );
         });
     };
 
     if (items && items.length > 0) {
-        const ContentWrapper = mobile ? MobileWrapper : Wrapper;
         return (
             <>
-                <ContentWrapper>{renderItems(mobile)}</ContentWrapper>
+                {mobile ? (
+                    <MobileWrapper ref={ref}>
+                        {renderItems(mobile)}
+                    </MobileWrapper>
+                ) : (
+                    <Wrapper ref={ref}>
+                        {renderItems(mobile)}
+                        {items &&
+                            selectedIndex >= 0 &&
+                            items[selectedIndex] &&
+                            toggleDropdown && (
+                                <Menu
+                                    items={items[selectedIndex].subMenu}
+                                    selectedId={selectedId}
+                                    mobile={false}
+                                    onItemClick={handleSubLinkClick(
```

Comment:
actually since menu also has an `onItemClick` that passes the event and selected item, could `handleSubLinkClick` simply take in those arguments instead?